ui/packages/catalog-ui-search/src/main/webapp/component/builder/builder.view.js

```diff
@@ -0,0 +1,214 @@
+                } else if(this.isMultipleAvailableTypes()) {
+                    this.showMetacardTypeSelection();
+                } else {
+                    this.handleSystemTypes();
+                }
+            },
+            handleSelectedAvailableType() {
+                this.$el.removeClass('is-selecting-available-types');
+
+                const selectedAvailableType = this.builderAvailableType.currentView.model.getValue()[0];
+
+                this.model.set('selectedAvailableType', selectedAvailableType);
+
+                this.showMetacardBuilder();
+            },
+            showMetacardBuilder() {
+
+                const selectedAvailableType = this.model.get('selectedAvailableType');
+
+                const metacardDefinition = metacardDefinitions.metacardDefinitions[selectedAvailableType.metacardType];
+
+                const propertyCollection = {
+                    'metacard-type': selectedAvailableType.metacardType
+                };
+
+                selectedAvailableType.visibleAttributes
+                    .filter((attribute) => !metacardDefinitions.isHiddenType(attribute))
+                    .filter((attribute) => !metacardDefinition[attribute].readOnly)
+                    .filter((attribute) => attribute !== "id")
+                    .forEach((attribute) => {
+                    if (metacardDefinition[attribute].multivalued) {
+                      propertyCollection[attribute] = [];
+                    } else if (metacardDefinitions.enums[attribute]) {
+                      propertyCollection[attribute] = metacardDefinitions.enums[attribute][0];
+                    } else {
+                      propertyCollection[attribute] = "";
+                    }
+                });
+
+                this.model.set('metacard', propertyCollection);
+                this.$el.addClass('is-building');
+
+            },
+            handleMetacard() {
+                const metacard = this.model.get('metacard');
+                this.builderProperties.show(PropertyCollectionView.generatePropertyCollectionView([metacard]));
+                this.builderProperties.currentView.$el.addClass("is-list");
+
+            },
+            onBeforeShow() {
+                    this.handleAvailableTypes();
+            },
+            edit() {
+                        this.$el.addClass('is-editing');
+                        this.builderProperties.currentView.turnOnEditing();
+                        this.builderProperties.currentView.focus();
+            },
+            cancel() {
+                        this.$el.removeClass('is-editing');
+                        this.builderProperties.currentView.revert();
+                        this.builderProperties.currentView.turnOffEditing();
+            },
+            save() {
+                        this.$el.removeClass('is-editing');
+
+                        const editedMetacard = this.builderProperties.currentView.toPropertyJSON([], []);
```

Comment:
❓ Why do we pass in the blank array arguments?

Reply:
removed